### ISSUE TITLE
vim or nvim window update lag

### ISSUE THREAD

**Issue by donaldcallen** [conformal/spectrwm#215]
Either vim or nvim in a terminal window (st or qterminal) exhibit noticeable lags when doing operations that require updates to the window. Really annoyingly slow. Doing the same with dwm on the same system (up-to-date Arch Linux), same terminal emulators, does not exhibit this behavior.

**Comment by mschilli87** [comment]
I use nvim in st in spectrw on several machines with different distrubutions for several years, before the same with vim and I never experienced this. Can you share some more details (versions, TERM, vim plugins etc.) and a step-by-step guide how to reproduce the issue?

**Comment by donaldcallen** [comment]
On Tue, 18 Sep 2018 at 10:32, Marcel Schilling <notifications@github.com>
wrote:

> I use nvim in st in spectrw on several machines with different
> distrubutions for several years, before the same with vim and I never
> experienced this. Can you share some more details (versions, TERM, vim
> plugins etc.) and a step-by-step guide how to reproduce the issue?
>

st version is 0.8.1, though I have seen the same behavior with qterminal as
well. Running st, TERM is  st-256color.

.vimrc:
set guifont=Monospace\ Bold\ 8
set nocp
set shiftwidth=4
set tabstop=4
set expandtab
set background=light
set lispwords=define,let,let*,do,lambda,case
execute pathogen#infect()
filetype on
filetype plugin on
autocmd FileType * setlocal formatoptions-=c formatoptions-=r
formatoptions-=o
let g:yankring_persist = 0
let g:yankring_share_between_instances = 0
let g:yankring_history_dir = '$VIM'
set noincsearch
set nohlsearch
syntax off
let g:sqlutil_align_where = 0
let g:sqlutil_align_comma = 1
map <f2> a <Esc>
map <f3> a,<Esc>
map <f4> a)<Esc>
map <f5> a_<Esc>
map <M-f> :set formatoptions=
:nmap <Space> i_<Esc>r
:nmap <Tab> a_<Esc>r

The machine is a Lenovo M600, 8 GB, 256GB SSD. This is my primary system,
but I have others. I will try comparing editing under spectrwm vs dwm on
another machine or two, to see if this issue is machine-specific. I'll
report back.

On the Lenovo machine, operations like ':e .' are just sluggish (screen
updates happen after a small but noticeable delay), whereas they are
instantaneous when running dwm. There is also an issue with deleted windows
persisting on the screen in spectrwm, which I've reported separately.

**Comment by donaldcallen** [comment]
I did a bit more on this -- checked the behavior on one of my Thinkpads. Responsiveness was normal, so the problem was specific to the M600, which led me to look at my .xinitrc and there was some old cruft in there  that was for an older machine that had the same hostname as the M600 (it's been awhile since I've tried spectrwm). I removed it, restarted X and vim/nvim performance seems fine now. So this issue was due to my mis-configuration and therefore I'm closing it.
